src/components/FoldableText/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+import { Flex, Text, ChevronDownIcon, ChevronUpIcon } from '@pancakeswap-libs/uikit'
+import useI18n from 'hooks/useI18n'
+
+const Wrapper = styled(Flex)`
+  cursor: pointer;
+`
+
+const StyledExpandButton = styled(Flex)`
+  svg {
+    margin-top: 2px;
+    fill: ${({ theme }) => theme.colors.primary};
+  }
+`
+
+const StyledChildrenFlex = styled(Flex)<{ isExpanded?: boolean }>`
+  overflow: hidden;
+  height: ${({ isExpanded }) => (isExpanded ? '100%' : '0px')};
+  padding-bottom: ${({ isExpanded }) => (isExpanded ? '16px' : '0px')};
+  border-bottom: 1px solid ${({ theme }) => theme.colors.inputSecondary};
+`
+
+const FoldableText: React.FC<{ title?: string }> = ({ title, children }) => {
+  const [isExpanded, setIsExpanded] = useState(false)
+  const TranslateString = useI18n()
+
+  return (
+    <Wrapper flexDirection="column" onClick={() => setIsExpanded(!isExpanded)} mt="24px">
```

Comment:
Will this component always need 24px margin top? If this is a more generic component lets pass down the rest of the props and use it like so

```
<FoldableText mt="24px" />
```

Reply:
Nice addition man, I've made the wrapper receive `...props` & its types are extending `BoxProps`